SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimPhase2Producer.cc

```diff
@@ -0,0 +1,185 @@
+
+  algo_ = std::make_unique<EcalEBPhase2TrigPrimAlgo>(&eTTmap, &theGeometry, binOfMaximum_, debug_);
+
+  // get a first version of the records
+  cacheID_ = this->getRecords(setup);
+
+  nEvent_ = 0;
+}
+
+unsigned long long EcalEBTrigPrimPhase2Producer::getRecords(edm::EventSetup const& setup) {
+  // get parameter records for xtals
+  auto theEcalEBPhase2TPGLinearization_handle = setup.getHandle(theEcalEBPhase2TPGLinearization_Token_);
+  const EcalEBPhase2TPGLinearizationConst* ecaltpLin = theEcalEBPhase2TPGLinearization_handle.product();
+  //
+  edm::ESHandle<EcalLiteDTUPedestalsMap> theEcalTPGPedestals_handle = setup.getHandle(theEcalTPGPedestals_Token_);
+  const EcalLiteDTUPedestalsMap* ecaltpPed = theEcalTPGPedestals_handle.product();
+  //
+  // auto theEcalEBPhase2TPGPedestals_handle = setup.getHandle(theEcalEBPhase2TPGPedestals_Token_);
+  //const EcalEBPhase2TPGPedestalsMap* ebTPPedestals = theEcalEBPhase2TPGPedestals_handle.product();
+  //
+  edm::ESHandle<EcalTPGCrystalStatus> theEcalTPGCrystalStatus_handle = setup.getHandle(theEcalTPGCrystalStatus_Token_);
+  const EcalTPGCrystalStatus* ecaltpgBadX = theEcalTPGCrystalStatus_handle.product();
+  //
+  edm::ESHandle<EcalEBPhase2TPGAmplWeightIdMap> theEcalEBTPGAmplWeightIdMap_handle =
+      setup.getHandle(theEcalEBTPGAmplWeightIdMap_Token_);
+  const EcalEBPhase2TPGAmplWeightIdMap* ecaltpgAmplWeightMap = theEcalEBTPGAmplWeightIdMap_handle.product();
+  //
+  edm::ESHandle<EcalEBPhase2TPGTimeWeightIdMap> theEcalEBTPGTimeWeightIdMap_handle =
+      setup.getHandle(theEcalEBTPGTimeWeightIdMap_Token_);
+  const EcalEBPhase2TPGTimeWeightIdMap* ecaltpgTimeWeightMap = theEcalEBTPGTimeWeightIdMap_handle.product();
+  //
+  edm::ESHandle<EcalTPGWeightGroup> theEcalTPGWeightGroup_handle = setup.getHandle(theEcalTPGWeightGroup_Token_);
+  const EcalTPGWeightGroup* ecaltpgWeightGroup = theEcalTPGWeightGroup_handle.product();
+  //
+  //edm::ESHandle<EcalTPGTowerStatus> theEcalTPGTowerStatus_handle = setup.getHandle(theEcalTPGTowerStatus_Token_);
+  //const EcalTPGTowerStatus* ecaltpgBadTT = theEcalTPGTowerStatus_handle.product();
+  //
+  //edm::ESHandle<EcalTPGSpike> theEcalTPGSpike_handle = setup.getHandle(theEcalTPGSpike_Token_);
+  //const EcalTPGSpike* ecaltpgSpike = theEcalTPGSpike_handle.product();
+
+  ////////////////
+  algo_->setPointers(ecaltpPed, ecaltpLin, ecaltpgBadX, ecaltpgAmplWeightMap, ecaltpgTimeWeightMap, ecaltpgWeightGroup);
+
+  return setup.get<EcalEBPhase2TPGLinearizationConstRcd>().cacheIdentifier();
+  //  return setup.get<EcalLiteDTUPedestalsRcd>().cacheIdentifier();
+}
+
+void EcalEBTrigPrimPhase2Producer::endRun(edm::Run const& run, edm::EventSetup const& setup) { algo_.reset(); }
+
+EcalEBTrigPrimPhase2Producer::~EcalEBTrigPrimPhase2Producer() {}
+
+// ------------ method called to produce the data  ------------
+void EcalEBTrigPrimPhase2Producer::produce(edm::Event& e, const edm::EventSetup& iSetup) {
+  nEvent_++;
```

Comment:
Is this variable meaningful when running multithreaded?

Reply:
I do not know if it's useful when running multithreaded but is useful to me when I make my tests

Reply:
If you run with multiple streams in your job, your debugging printout will have multiple outputs with the same value for nEvent_. I'd suggested you are much better off switching from `std::cout` to the MessageLogger as the message logger is safe while running multi-threaded (unlike `std::cout` which gets its messages garbled) and the MesageLogger would tell you which event the message comes from.